src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -48,7 +48,7 @@ public RepoConfiguration(RepoLocation location, String branch) {
     public RepoConfiguration(RepoLocation location, String branch, List<Format> formats, List<String> ignoreGlobList,
             boolean isStandaloneConfigIgnored, List<CommitHash> ignoreCommitList) {
         this.location = location;
-        this.branch = branch;
+        this.branch = location.isEmpty() ? "" : branch;
```

Comment:
Possible to use `DEFAULT_BRANCH` instead of `""`? 
Rationale is that, in the unforeseen event that we perform a `git checkout ""`, it will throws an error.